tree/ntuple/v7/inc/ROOT/RColumn.hxx

```diff
@@ -249,14 +231,7 @@ public:
 
    template <typename CppT>
    CppT *MapV(const RClusterIndex &clusterIndex, NTupleSize_t &nItems) {
-      if (!fReadPage.Contains(clusterIndex)) {
-         MapPage(clusterIndex);
-      }
-      // +1 to go from 0-based indexing to 1-based number of items
-      nItems = fReadPage.GetClusterRangeLast() - clusterIndex.GetIndex() + 1;
-      return reinterpret_cast<CppT*>(
-         static_cast<unsigned char *>(fReadPage.GetBuffer()) +
-         (clusterIndex.GetIndex() - fReadPage.GetClusterRangeFirst()) * RColumnElement<CppT>::kSize);
+      return MapV<CppT>(GetGlobalIndex(clusterIndex), nItems);
```

Comment:
Does this make a speed difference? I'd assume it makes things (slightly) slower because we test two times if another pages needs to be mapped.

Reply:
Correct, there are two calls to `RPage::Contains` now. I'm not sure how expensive they are since the overload `Contains(NTupleSize_t globalIndex)` is literally just `return (globalIndex >= fRangeFirst) && (globalIndex < fRangeFirst + NTupleSize_t(fNElements));`. Do you have a suggestion what tests / benchmarks to run? We could think of having a private implementation method that assumes the right page has already been mapped in, then we could avoid it.

Reply:
I indeed saw some negative impact on the ATLAS analysis benchmark from https://github.com/jblomer/iotools. While investigating, I found https://github.com/root-project/root/pull/14406 which improves the performance more than would have degraded here. Additionally it already avoids the stack allocation in the recursive `RColumn::ReadV`, so the changes in `RColumn.hxx` are not a prerequisite anymore for this limit testing PR. I removed the commits for now, but I still think we should avoid the code duplication and avoid the recursive implementation - for another time.